src/RenderDocs.hs

```diff
@@ -37,16 +37,33 @@ saveDocsForEnvs ctx pathsAndEnvBinders =
   let dir = projectDocsDir ctx
       title = projectTitle ctx
       generateIndex = projectDocsGenerateIndex ctx
-      allEnvNames = fmap (getModuleName . fst . getEnvAndMetaFromBinder . snd) pathsAndEnvBinders
+      dependencies = getDependenciesForEnvs (Prelude.map (\(p, b) -> (p, fst (getEnvAndMetaFromBinder b))) pathsAndEnvBinders)
+      pathsAndEnvBinders' = pathsAndEnvBinders ++ dependencies
+      allEnvNames = fmap fst pathsAndEnvBinders'
    in do
-        mapM_ (saveDocsForEnvBinder ctx allEnvNames) pathsAndEnvBinders
+        mapM_ (saveDocsForEnvBinder ctx allEnvNames) pathsAndEnvBinders'
         when
           generateIndex
           ( writeFile
               (dir </> title ++ "_index.html")
               (projectIndexPage ctx allEnvNames)
           )
         putStrLn ("Generated docs to '" ++ dir ++ "'")
+  where
+    getDependenciesForEnvs = Prelude.concat . Prelude.map getEnvDependencies
```

Comment:
If you want, you could also import `Env` and likely use some combination of `findModules`, `envPublicBindingNames`, or something similar for the getDependencies part.